Refine CONTRIBUTING.md (#12627)
JabRef/jabref: CONTRIBUTING.md

We welcome contributions to JabRef and encourage you to follow the GitHub workflow specified below.
If you are not familiar with this type of workflow, take a look at GitHub's excellent overview on the [GitHub flow](https://docs.github.com/en/get-started/using-github/github-flow) and the explanation of [Feature Branch Workflow](https://atlassian.com/git/tutorials/comparing-workflows#feature-branch-workflow) for the idea behind this kind of development.

> **JabRef regards contributors as [software developers not as programmers](https://www.phoenix.edu/blog/programmer-vs-software-engineer-key-differences.html)**.

Reviewer: ```suggestion
**JabRef regards contributors as [software engineers, not as programmers](https://www.phoenix.edu/blog/programmer-vs-software-engineer-key-differences.html)**.
```

Author: I re-iterated on "software engineers" vs. "software developers". I think, its good, to update there. Even if "software engineer" might be too much; but we require (sometimes) more than a "software developer"

### I am a student and I want to start with something easy

We collect good issues to start with at our [list of unassigned good first issues](https://github.com/JabRef/jabref/issues?q=sort%3Aupdated-desc%20is%3Aissue%20is%3Aopen%20label%3A%22good%20first%20issue%22%20no%3Aassignee).

### I am a student and I want to choose from a curated list of university projects

> Additionally to "good first issues", we offer curated issues to work from.

Reviewer: ```suggestion
Apart from "good first issues", we also offer curated issues to work on.
```
OR
```suggestion
Apart from "good first issues", we also offer collections of curated issues to work on.
```

> We categorize them in "small", "medium", and "large".

Reviewer: ```suggestion
We categorize them as "small", "medium", and "large".
```
OR
```suggestion
We categorize them into "small", "medium", and "large".
```

All of them are intended to
a) bring you closer to a larger code base with a dedicated issue and
b) be clear in their description of work.
Sometimes, you need to refine the requirements:

> We see contributors as software developers and [not as programmers](https://www.phoenix.edu/blog/programmer-vs-software-engineer-key-differences.html).

Reviewer: ```suggestion
We see contributors as software engineers and [not just programmers](https://www.phoenix.edu/blog/programmer-vs-software-engineer-key-differences.html).
```

> Thus, you really need work on requirements refinement.

Reviewer: ```suggestion
Thus, requirement engineering inherently comes up as a part of the process.
```

Take a look at [JabRef's candidates for university projects](https://github.com/orgs/JabRef/projects/3).

> Here, you get a table of curated issues.

Reviewer: ```suggestion
Here, you will get a table of curated issues.
```

The table offers filtering for small, medium, and large projects.
You can check the main focus (UI, logic, or both), the issue understanding effort, the implementation effort, and testing effort.

> The "issue understanding effort" is an indication how much time you will need to issue before you can do any coding.

Reviewer: ```suggestion
The "issue understanding effort" is an indication of how much time you will need to understand the issue before you can do any coding. It may depend on how easy it is to reproduce the issue, how much background knowledge is needed, etc.
```
